Documentation/dev-tools/kfence.rst

```diff
@@ -211,26 +224,24 @@ Implementation Details
 ----------------------
 
 Guarded allocations are set up based on the sample interval. After expiration
-of the sample interval, a guarded allocation from the KFENCE object pool is
-returned to the main allocator (SLAB or SLUB). At this point, the timer is
-reset, and the next allocation is set up after the expiration of the interval.
-To "gate" a KFENCE allocation through the main allocator's fast-path without
-overhead, KFENCE relies on static branches via the static keys infrastructure.
-The static branch is toggled to redirect the allocation to KFENCE.
+of the sample interval, the next allocation through the main allocator (SLAB or
+SLUB) returns a guarded allocation from the KFENCE object pool. At this point,
+the timer is reset, and the next allocation is set up after the expiration of
+the interval.  To "gate" a KFENCE allocation through the main allocator's
+fast-path without overhead, KFENCE relies on static branches via the static
+keys infrastructure.  The static branch is toggled to redirect the allocation
+to KFENCE.
 
 KFENCE objects each reside on a dedicated page, at either the left or right
 page boundaries selected at random. The pages to the left and right of the
 object page are "guard pages", whose attributes are changed to a protected
 state, and cause page faults on any attempted access. Such page faults are then
 intercepted by KFENCE, which handles the fault gracefully by reporting an
-out-of-bounds access. The side opposite of an object's guard page is used as a
-pattern-based redzone, to detect out-of-bounds writes on the unprotected sed of
-the object on frees (for special alignment and size combinations, both sides of
-the object are redzoned).
+out-of-bounds access.
 
-KFENCE also uses pattern-based redzones on the other side of an object's guard
-page, to detect out-of-bounds writes on the unprotected side of the object;
-these are reported on frees.
+KFENCE also uses pattern-based redzones on the opposite side of an object's
```

Comment:
Still unclear. Logically, a redzone should be on the other side of an object. And the text says that a redzone is on the other side of a guard page. I guess both are true, but these two redzones are guarding two different objects. I'd rather refer to `the other side of an object`.

Reply:
Not sure how to put this more succinctly, but it's technically: object xor page => redzone. The figure illustrates what we have best.

If this text here somehow mismatches what you infer from the figure, it needs to be fixed until it reads right. :-) I'll try again.